src/makers/win32/appx.js

```diff
@@ -65,7 +65,10 @@ export default async (dir, appName, targetArch, forgeConfig, packageJSON) => { /
     opts.devCert = await createDefaultCertificate(opts.publisher, outPath);
   }
 
-  if (!opts.publisher.match(/^CN=/)) {
+  if (typeof opts.publisher.match !== 'function') {
+    // Checking for .match prevents an error when opts.publisher is not a string
+    opts.publisher = `CN=${opts.publisher.name || opts.publisher.email || opts.publisher.url || opts.publisher}`;
```

Comment:
I cannot say for certain (I'm not an X.509 expert by any means), but if the previous format was correct (assuming a string with the author's name) then if it's an object with a `name` property instead that should maintain proper operation. In case `name` is falsey it attempts to try email and then url and finally whatever the value of the object itself is. There may be a better solution than this, but my rational was that this should at least yield some sensible output that does not generate an error.